### PR TITLE
add elfio package

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/help-c-cpp @conda-forge/staged-recipes

**Comment by awvwgk** [comment]
Thanks.

**Comment by matthiasdiener** [comment]
> Thanks. What is your opinion on using CMake rather than autoconf for the build?

I think it's a good idea and am preparing a patch.

**Comment by matthiasdiener** [comment]
What do you think of the recipe @awvwgk ?

**Comment by matthiasdiener** [comment]
A gentle ping for @conda-forge/staged-recipes for a review

File: recipes/elfio/bld.bat [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cmake -DCMAKE_INSTALL_PREFIX=%PREFIX% -G Ninja -B _build
+
+cmake --install _build
```
Comment (ocefpaf) [inline recipes/elfio/bld.bat:3]: Can we run ctest on Windows too?

Comment (matthiasdiener) [reply recipes/elfio/bld.bat:3]: No, the tests don't seem to build on Windows due to some boost linking errors. This is a header-only library anyway.

**Pull request merged**